src/lender/admin/pool.sol

```diff
@@ -45,92 +71,231 @@ contract PoolAdmin is Auth {
         seniorMemberlist.updateMember(usr, validUntil);
         emit UpdateSeniorMember(usr, validUntil);
     }
 
-    function updateSeniorMembers(address[] memory users, uint256 validUntil) public admin {
+    function updateSeniorMembers(address[] memory users, uint256 validUntil) public level1 {
         seniorMemberlist.updateMembers(users, validUntil);
         emit UpdateSeniorMembers(users, validUntil);
     }
 
-    function updateJuniorMember(address usr, uint256 validUntil) public admin {
+    function updateJuniorMember(address usr, uint256 validUntil) public level1 {
         juniorMemberlist.updateMember(usr, validUntil);
         emit UpdateJuniorMember(usr, validUntil);
     }
 
-    function updateJuniorMembers(address[] memory users, uint256 validUntil) public admin {
+    function updateJuniorMembers(address[] memory users, uint256 validUntil) public level1 {
         juniorMemberlist.updateMembers(users, validUntil);
         emit UpdateJuniorMembers(users, validUntil);
     }
     
+    // --- Risk Management, authorized by level 2 admins ---
+    event OverrideWriteOff(uint loan, uint writeOffGroupIndex);
+    event AddRiskGroup(uint risk_, uint thresholdRatio_, uint ceilingRatio_, uint rate_, uint recoveryRatePD_);
+    event AddRiskGroups(uint[] risks_, uint[] thresholdRatios_, uint[] ceilingRatios_, uint[] rates_);
+    event AddWriteOffGroup(uint rate_, uint writeOffPercentage_, uint overdueDays_);
+    event SetMatBuffer(uint value);
+    event UpdateNFTValue(bytes32 nftID_, uint value);
+    event UpdateNFTValueRisk(bytes32 nftID_, uint value, uint risk_);
+    event UpdateNFTMaturityDate(bytes32 nftID_, uint maturityDate_);
+
+    function overrideWriteOff(uint loan, uint writeOffGroupIndex_) public level2 {
+        navFeed.overrideWriteOff(loan, writeOffGroupIndex_);
+        emit OverrideWriteOff(loan, writeOffGroupIndex_);
+    }
+
+    function addRiskGroup(uint risk_, uint thresholdRatio_, uint ceilingRatio_, uint rate_, uint recoveryRatePD_) public level2 {
+        navFeed.file("riskGroup", risk_, thresholdRatio_, ceilingRatio_, rate_, recoveryRatePD_);
+        emit AddRiskGroup(risk_, thresholdRatio_, ceilingRatio_, rate_, recoveryRatePD_);
+    }
+
+    function addRiskGroups(uint[] memory risks_, uint[] memory thresholdRatios_, uint[] memory ceilingRatios_, uint[] memory rates_, uint[] memory recoveryRatePDs_) public level2 {
+        require(risks_.length == thresholdRatios_.length && thresholdRatios_.length == ceilingRatios_.length && ceilingRatios_.length == rates_.length, "non-matching-arguments");
+        for (uint i = 0; i < risks_.length; i++) {
+            addRiskGroup(risks_[i], thresholdRatios_[i], ceilingRatios_[i], rates_[i], recoveryRatePDs_[i]);
+        }
+    }
+
+    function addWriteOffGroup(uint rate_, uint writeOffPercentage_, uint overdueDays_) public level2 {
+        navFeed.file("writeOffGroup", rate_, writeOffPercentage_, overdueDays_);
```

Comment:
Did you double check the order of the parameters, if they are all uint like here?